src/components/transaction-dialog/transaction-dialog.js

```diff
@@ -0,0 +1,72 @@
+import angular from "angular";
+import template from "./transaction-dialog.html";
+
+import ngAnimate from "angular-animate";
+import ngMaterial from "angular-material";
+
+export class TransactionDialogController {
+  constructor($mdDialog, $filter) {
+    this.$mdDialog = $mdDialog;
+    this.$filter = $filter;
+    this.tags = ["transport", "food", "salary"];
+    this.accounts = ["Privat", "BoaBank", "Cash"];
+  }
+
+  $onChanges(changes) {
+    if (changes.isOpen.currentValue) {
+      this.showDialog(changes.isOpen.currentValue.event);
+    }
+  }
+
+  showDialog(ev) {
+    this.$mdDialog.show({
+      template,
+      controllerAs: "$ctrl",
+      controller: DialogController,
+      parent: angular.element(document.body),
+      targetEvent: ev,
```

Comment:
do you need this target event?

Reply:
it is a target from where dialog shows. For now it is for animation, but in future (in my opinion) I will need it to know about what data to edit

Reply:
ok. maybe I will not need it in future. but md-material uses it for dialog animation

Reply:
![image](https://user-images.githubusercontent.com/14272396/38209903-6cb024c6-36be-11e8-8355-3a0791845d2a.png)

Reply:
ok